### ISSUE TITLE
Tests still require nose

### ISSUE THREAD

**Issue by TheChymera** [datalad/datalad#7100]
even though the test suite now works with pytest, nose is needed for testing, and not just for the docs (which is what `setup.py` indicates):

```console
chymera@decohost ~/src/datalad $ git rev-parse HEAD
b3ff763c8f1eff6ec6c21c837dd4c37136944fe0
chymera@decohost ~/src/datalad $ grep nose datalad/tests/utils.py
from nose import SkipTest
from nose.plugins.attrib import attr
from nose.tools import (
from nose.tools import assert_in
from nose.tools import assert_in as in_
from nose.tools import (
    from nose.ext import dtcompat
    from nose.plugins import (
    # Monkey patch nose so it does not ERROR out whenever code asks for fileno
    # of the output. See https://github.com/nose-devs/nose/issues/6
        from nose import SkipTest
    The only way would be to instruct testing framework (i.e. nose
def ignore_nose_capturing_stdout(func):
    Originally was intended as a decorator workaround for nose's behaviour
    with redirecting sys.stdout, but now we monkey patch nose now so no test
    https://code.google.com/p/python-nose/issues/detail?id=243&can=1&sort=-id&colspec=ID%20Type%20Status%20Priority%20Stars%20Milestone%20Owner%20Summary
        "@ignore_nose_capturing_stdout no longer does anything - nose should "
 ```

**Comment by bpoldrack** [comment]
@TheChymera: Is it just that you saw `nose` being used there or that you could not run the tests without `nose` being installed?

The intention here was to keep a nose-dependent `tests/utils.py` for a while to provide it for extensions that may still rely on nose (including our own extensions that could not be switch all at once). The tests in `datalad` should actually not use anything in `tests/utils.py` anymore, but the equivalents from `tests/utils_pytest.py`, so that `tests/utils.py` can be deleted sometime in the not too distant future.

**Comment by bpoldrack** [comment]
I'm closing this, since I think there's nothing to do ATM.

@TheChymera: Please feel free to reopen, if you think I misunderstood the point of it.

**Comment by TheChymera** [comment]
The tests actually failed, because the code was loaded when running the test suite, but it's ok, I just listed nose as a dependency, will re-open if it's still needed by the next release.

**Comment by bpoldrack** [comment]
Actually, that suggests there's something incomplete on our end.
This should be excluded from inspection by pytest. Plus: Since CI isn't failing here, we probably manage to still install `nose` in their environment somehow.

Thx, @TheChymera

**Comment by bpoldrack** [comment]
Fixed by #7136
